W-13198191-add-ignored-files-Studio-CP (#236)
mulesoft/docs-munit: modules/ROOT/pages/coverage-studio-concept.adoc

| *Processors*         | Total amount of processors inside the container
| *Coverage*            | Percentage of the event processors inside the container executed in the MUnit test
|===

== Disable Coverage Reports

If you choose so, you can disable the coverage calculation from your project run configuration so you test your flow without a coverage report.

Right-click *Package Explorer* and choose *Run As* > *Run Configurations*. Select your MUnit run configuration file to access your suite configuration, and deselect the *Run coverage* option in your coverage settings.

image::disable-munit-coverage-in-anypoint-studio.png["MUnit run and Run coverage highlighted"]

== Exclude XML Files from MUnit Coverage Report

You can exclude one or more XML files from an MUnit Coverage report when testing your application. 

. Go to the flow to change.
. Click *Run* > *Run Configuration*.
. Click the name of your project.
. In the *Ignored files* section, click *Add*.
+
image::add-ignored-files.png["Add Ignored File window highlighted in the Test tab"]
+
. Type the name of the XML file to exclude from the Coverage report and click *OK*.
. Click *Apply* and *Run*.

> == See Also

Reviewer: Add a link to this one too?
https://docs.mulesoft.com/munit/latest/munit-coverage-report

Author: Sure! Thanks!

> You can run a coverage report from Anypoint Studio.

Reviewer: What is a coverage report and why would you run one?

Author: The overview of the coverage-report purpose is explained at: https://docs.mulesoft.com/munit/latest/munit-coverage-report. The coverage-studio-concept.adoc article is a child of the munit-coverage-report.adoc article.

Reviewer: OK! Presumably linked from this one, too.

Author: Yes! The parent takes the reader to the two other articles.
Thanks!

The following configurations apply only when you execute your MUnit tests in Studio. They do not apply when running tests from Maven in a CI/CD environment. +
To run coverage reports from Maven see xref:coverage-maven-concept.adoc[].

== Enable Coverage Reports in Anypoint Studio

> The Coverage button enables you to see which flows, and which percentage of event processors in those flows the test covers:

Reviewer: Consider revising to focus on the user task instead of the UI.

Author: Sure! I'll rewrite the focus.

image::using-munit-coverage-in-anypoint-studio.png["MUnit Coverage tab highlighted in the MUnit section."]

The Overall Coverage value represents the percentage of the Mule application event processors that the MUnit test executes.

> You can click on *Generate Report* to get the report that details the flows and event processors in the Mule configuration files that are included in the MUnit test.

Reviewer: Good rewrite!

Author: Thanks!

> You can click on *Generate Report* to get the report that details the flows and event processors in the Mule configuration files that are included in the MUnit test.

Reviewer: ```suggestion
Click *Generate Report* to get the report that details the flows and event processors in the Mule configuration files that are included in the MUnit test.
```

Reviewer: See Style Guide: https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-%22click%22

Author: OK! Thanks!

image::munit-coverage-report-in-anypoint-studio.png["Files included in the report are highlighted."]

[%header,cols="20a,80a"]
|===
| Column      | Description

> | *Resource*    | The name of the Mule Configuration file

Reviewer: ```suggestion
| *Resource*    | Name of the Mule configuration file
```

Author: OK! Thanks!

> | *Containers* | Number of flows inside the Mule configuration file

Reviewer: ```suggestion
| *Containers* | Number of flows inside the Mule configuration file
```

Reviewer: (I just looked up in vs inside because I always default to in. I still don't understand the nuances.)
https://www.differencebetween.com/difference-between-in-and-inside/

Author: For me, "inside" seems to be an emphasis on "in" hahahahah

Reviewer: I asked Diane and she didn't know the difference. I always use "in" because it's shorter and costs less to translate :>

Author: Good to know! Thanks!

> | *Weight*      | The percentage of your application in a configuration file. +

Reviewer: Check the SG for table punctuation rules.
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-tables

Author: Sure! Thanks!

>                 It is calculated as a percentage of the overall number of event processors in the documentation and the

Reviewer: I prefer to avoid "It is" construction, but it's not against our style. Suggest "This value is".

Author: OK! Thanks!

>                 amount of event processors in the Mule configuration file.

Reviewer: ```suggestion
                number of event processors in the Mule configuration file.
```

Author: OK! Thanks!

> | *Coverage*    | The percentage of the event processors inside the Mule Configuration file that is executed in the MUnit test

Reviewer: ```suggestion
| *Coverage*    | The percentage of event processors in the Mule configuration file that the MUnit test executes
```

Reviewer: Active voice (Not sure if that changes the meaning.)

Author: I think the meaning is OK. Thanks!

|===

> Additionally, you can click on each *Resource* to get a more specific and granular report of the Event Processors executed in that resource:

Reviewer: ```suggestion
Additionally, you can click each *Resource* to get a more specific and granular report of the Event Processors executed in that resource:
```

Author: OK! Thanks!

> image::munit-granular-coverage-report-in-anypoint-studio.png["Granular report is highlighted."]

Reviewer: Maybe something like "Xxx with the granular report highlighted."

Author: OK! Thanks!

[%header,cols="20a,80a"]
|===
| Column              | Description

> | *Name*                | The name of the container

Reviewer: I prefer to leave out the article (The) for easier scanning.

Author: OK! Thanks!

| *Type*                | The type of the container

> | *Covered Processors* | The processors inside the container that is executed during the MUnit test

Reviewer: ```suggestion
| *Covered Processors* | Processors inside the container that are executed during the MUnit test
```

Reviewer: Check subject-verb agreement. Kim has this in the edit tips:
https://confluence.internal.salesforce.com/display/MTDT/Edit+Tips+Archive#EditTipsArchive-Checknumericalsubject-verbagreement

Author: Sure! Thanks!

| *Processors*         | The total amount of processors inside the container
| *Coverage*            | The percentage of the event processors inside the container executed in the MUnit test
|===

== Disable Coverage Reports

> If you choose so, you can disable the coverage calculation from your project run configuration.

Reviewer: Why would you disable it? Give an example.

Author: It may be the case that developers want to test their flow but they do not want to have a coverage report, so they can disable this feature. This feature is not mandatory for testing a flow.

Reviewer: Again, good information to share with the user.

Author: OK! Thanks!

> Right-click in your *Package Explorer*, and choose *Run As* and *Run Configurations...*. Select your MUnit run configuration file to access your suite configuration, and deselect the *Run coverage* option in your coverage settings.

Reviewer: ```suggestion
Right-click *Package Explorer* and choose *Run As* > *Run Configurations*. Select your MUnit run configuration file to access your suite configuration, and deselect the *Run coverage* option in your coverage settings.
```

Reviewer: Not sure if that's correct. Is it a series of menu options?

Don't include the punctuation in UI elements.
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-UIelements

Author: I don't think I understand your comment. What is a series of menu options?
Thanks for your suggestions!

Reviewer: Is this a series of menu selections:
*Package Explorer*, and choose *Run As* and *Run Configurations...*

Author: Oh! I got it now! Yes! It's a menu that opens when you hover over the element of the menu.

> image::disable-munit-coverage-in-anypoint-studio.png["MUnit run and Run coverage highlighted."]

Reviewer: (Global) Remove terminating punctuation if the alt text isn't a complete sentence.

Author: OK! Thanks!

== Exclude XML Files from MUnit Coverage Report

> You can exclude one or more XML files from an MUnit Coverage report when testing your application. 

Reviewer: Why would you do that?

Reviewer: Should this be "MUnit coverage report" or "MUnit Coverage Report"? (i.e., is it a proper noun?)

Author: "MUnit Coverage Report" is a proper name and this feature is "weird".
It may be the case that developers included third-party XML files in their projects. They may need to run a report that tests their XML files so they want to exclude third-party XML files. They may want to do this because the third-party XML files do not comply with the standards, or because of any other reason you can think of (for example, they may even exclude their own XML files).

Reviewer: These are good things to add to the short description (first paragraph) of the section. Give the user context as to why they should care.

Author: I think I'm lost. If needed, I may go over the explanation provided at https://docs.mulesoft.com/munit/latest/coverage-studio-concept, which is a different article.
Do you agree? In that article, maybe we should improve the definition of the "MUnit Coverage feature".

Reviewer: I'm confused. Isn't coverage-studio-concept this topic?

All I'm saying is that we need to give context to the users on why they would do the task. Examples, use cases, reasons for doing the thing. This page gives more info on that:
https://confluence.internal.salesforce.com/x/5LmeBw

We are going to be focusing more on the WHY and less on the UI.

Reviewer: I'm suggesting that you add some context - why would they exclude files? Something like "such as third-party XML files do not comply with standards".

Author: Oh! OK! Thanks!

> . Go to the flow you want to change.

Reviewer: ```suggestion
. Go to the flow to change.
```

Author: OK! Thanks!

> . Click *Run* > *Run Configuration*.

Reviewer: You use this construction here!

Author: Yes! You're right!
Thanks!

. Click the name of your project.
. In the *Ignored files* section, click *Add*.
+
image::add-ignored-files.png["Add Ignored File window highlighted in the Test tab."]
+

> . Type the name of the XML file you want to exclude from the Coverage report.

Reviewer: ```suggestion
. Enter the name of the XML file to exclude from the Coverage report.
```

Author: OK! Thanks!

> . Click *OK*.

Reviewer: Any way to combine these steps?

Author: Sure! Thanks!
